doc/patch-author-guide.md

```diff
@@ -214,6 +214,15 @@ safe to access:
  	 * Flush inherited counters to the parent - before the parent
  	 * gets woken up by child-exit notifications.
 ```
+Notes:
+* `kpatch_shadow_alloc` initializes only shadow variable metadata. It
+  allocates variable storage via `kmalloc` with the `gfp_t` flags it is
+  given, but otherwise leaves the area untouched. Initialization of a shadow
+  variable is the responsibility of the caller.
+* As soon as `kpatch_shadow_alloc` creates a shadow variable, its presence
+  will be reported by `kpatch_shadow_get`. Care should be taken to avoid any
+  potential race conditions between a kernel thread that allocates a shadow
+  variable and concurrent threads that may attempt to use it.
```

Comment:
With this note, I was targeting the use-before-initialization race:
```
sv = kpatch_shadow_alloc(obj, "var", sizeof(int), GFP_KERNEL);
... registered but not initialized ...
*sv = 123;
```
where another thread could get a hit on kpatch_shadow_get and then use it before this one has initialized it.  (Applicable to both kpatch and livepatch models.)

I think the two-patch approach you proposed was addressing a potential race under the livepatch consistency model only:
* task1 - successfully patched allocation path, makes new shadow variables
* task2 - free path not patched yet, leaks those new shadow variables

Since the patch author guide is kpatch-specific right now, I only mentioned this first race.  I think that one is best fixed by modifying the shadow variable API to accept an already allocated (and initialized!) variable space.

Let me know if I'm mis-remembering that conversation :)